internal/dag/builder.go

```diff
@@ -209,11 +209,14 @@ func (b *builder) lookupSecret(m meta) *Secret {
 	s := &Secret{
 		Object: sec,
 	}
-	if b.secrets == nil {
-		b.secrets = make(map[meta]*Secret)
+	if s.isValid() {
```

Comment:
Please move the success case outside the condition

```go
if !s.isValid() {
    return nil
}
if b.secrets == nil {
    b.secrets = make(map[meta]*Secret)
}
b.secrets[s.toMeta()] = s
return s
```
ref: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88